SwiftMQTT/SwiftMQTT/Models/MQTTPacketFactory.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  MQTTPacketFactory.swift
+//  SwiftMQTT
+//
+//  Created by David Giovannini on 5/21/17.
+//  Copyright © 2017 Ankit. All rights reserved.
+//
+
+import Foundation
+
+struct MQTTPacketFactory {
+
+    let constructors: [MQTTPacketType : (MQTTPacketFixedHeader, Data)->MQTTPacket] = [
+        .connAck : MQTTConnAckPacket.init,
+        .subAck : MQTTSubAckPacket.init,
+        .unSubAck : MQTTUnSubAckPacket.init,
+        .pubAck : MQTTPubAck.init,
+        .publish : MQTTPublishPacket.init,
+        .pingResp : { h, _ in MQTTPingResp.init(header: h) }
+    ]
```

Comment:
I think we should use `switch` when going through enum cases because compiler can warn us if we miss any cases. So, I prefer converting this map into a function and switching over all the packet types as:

```Swift
func packet(for type: MQTTPacketType, header: MQTTPacketFixedHeader, data: Data) -> MQTTPacket {
    switch type {
        csae .connAck:  return MQTTConnAckPacket( ....
    }
```

What do you think?